### PR TITLE
[RUN-3490] Add explanation of user role and session behaviour

### PR DESCRIPTION
**Pull request by akkrishnakumar**

### CODE REVIEW

File: content/en/docs/refguide/modeling/security/app-security/user-roles.md [hunk 1]
```diff
@@ -22,6 +22,11 @@ To access user roles, open **App Security** > **User roles** tab:
 
 {{< figure src="/attachments/refguide/modeling/security/app-security/user-roles/user-roles-example.png" class="no-border" >}}
 
+{{% alert color="warning" %}}
+The effects of changes to user roles are not immediate. This is because the effects are not applied to active sessions, due to which the UI might show outdated pages, data. The effects are only applied either after users log out or the session times out automatically.
```
Comment (emirmx) [inline content/en/docs/refguide/modeling/security/app-security/user-roles.md]: Don't we refresh the roles when we re-validate the session? Or even after we refresh the roles in runtime, does user need to reload the page to see the differences?

**Comment by MarkvanMents** [comment]
@akkrishnakumar and @emirmx - is this ready for my review or should I wait until the question above is resolved?

**Comment by akkrishnakumar** [comment]
> @akkrishnakumar and @emirmx - is this ready for my review or should I wait until the question above is resolved?

Yes, please wait until the review is done. The reviewer will notify once it is done.

File: content/en/docs/refguide/modeling/security/app-security/user-roles.md (hunk 1 above)
Comment (akkrishnakumar) [reply content/en/docs/refguide/modeling/security/app-security/user-roles.md]: Umm... this is about changing the roles of a specific user. I'm not sure if the session is refreshed immediately. But in a clustered setup, I'm sure that not all sessions in the cache are up-to-date. So there can be discrepancies.

Comment (emirmx) [reply content/en/docs/refguide/modeling/security/app-security/user-roles.md]: New version looks fine.

**Comment by emirmx** [comment]
I reviewed from the runtime side and looks good.
@MarkvanMents

**Comment by MarkvanMents** [comment]
@akkrishnakumar and @emirmx 
I've proofread this in https://github.com/mendix/docs/pull/8294/commits/540a41ecbfb8a526600ac4f5c60d5738b88dba04
Approving. Please let me know if you are happy for this to be published.
Thanks for the PR.
Mark

**Review by MarkvanMents — APPROVED** [review]
Looks good. Thanks for the PR.
Let me know if you are happy for this to be published.

**Comment by akkrishnakumar** [comment]
> Looks good. Thanks for the PR. Let me know if you are happy for this to be published.

Yes please 🙂

**Pull request merged**